cmake/host-tools.cmake

```diff
@@ -27,23 +27,33 @@ else()
   endif()
 
   if(${west_version} VERSION_LESS ${MIN_WEST_VERSION})
+    if(WEST_PYTHON)
+      if(NOT (${WEST_PYTHON} STREQUAL ${PYTHON_EXECUTABLE}))
+        set(PYTHON_EXECUTABLE_OUT_OF_SYNC "Note:\n\
+  The Python version used by west is:  ${WEST_PYTHON}\n\
+  The Python version used by CMake is: ${PYTHON_EXECUTABLE}\n\
+  This might be correct, but please verify your installation.\n")
+      endif()
+    endif()
     message(FATAL_ERROR "The detected west version is unsupported.\n\
   The version was found to be ${west_version}:\n\
     ${item}\n\
   But the minimum supported version is ${MIN_WEST_VERSION}\n\
   Please upgrade with:\n\
-      pip3 install --upgrade west")
+      ${PYTHON_EXECUTABLE} -mpip install --upgrade west\n\n\
```

Comment:
Just whitespace nitpicking, but do we really need the `\n\n` if they are in sync?

Reply:
good point.
I was to focused on the extended information, so will move those to `PYTHON_EXECUTABLE_OUT_OF_SYNC`

Reply:
Fixed, and also improved the message when `west` is not installed.

Reply:
Minor nitpick - can this be `-m pip`? While I understand it works, I don't remember ever seeing `-mpip`, and for someone not at advanced level of Python, that may seem as some adhoc integral Python switch, instead of `-m` and its parameter.

Reply:
space added, so now: `-m pip install ...`